### PR TITLE
sas7bdat: Check if the SAS file has zero variables

### PR DESCRIPTION
**Pull request by AbdealiLoKo**
If the given SAS file has 0 rows, throw an error for the EmptyData file.
When reading, check that the column information is available. If not,
throw an error.

- [ ] closes #xxxx
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/io/sas/test_sas7bdat.py [hunk 1]
```diff
@@ -174,3 +176,9 @@ def test_date_time():
     df0 = pd.read_csv(fname, parse_dates=['Date1', 'Date2', 'DateTime',
                                           'DateTimeHi', 'Taiw'])
     tm.assert_frame_equal(df, df0)
+
+def test_zero_variables():
+    dirpath = tm.get_data_path()
```
Comment (jreback) [inline pandas/tests/io/sas/test_sas7bdat.py]: can you add the issue number (if no issue, then this PR number) as a comment

File: pandas/tests/io/sas/test_sas7bdat.py [hunk 2]
```diff
@@ -174,3 +176,9 @@ def test_date_time():
...
+    fname = os.path.join(dirpath, "zero_variables.sas7bdat")
+    with pytest.raises(EmptyDataError):
```
Comment (jreback) [inline pandas/tests/io/sas/test_sas7bdat.py]: can you test with the zero observations file as well (is it the same result)? you can parametrize if that'st the case

File: pandas/tests/io/sas/test_sas7bdat.py [hunk 3]
```diff
@@ -174,3 +176,9 @@ def test_date_time():
...
+        df = pd.read_sas(fname)
```
Comment (jreback) [inline pandas/tests/io/sas/test_sas7bdat.py]: don't need the df = here

**Review by jreback — CHANGES_REQUESTED** [review]
can you add a whatsnew note for 0.22 (can be in bug fixes). what did this do before with an empty file?

File: pandas/tests/io/sas/test_sas7bdat.py (hunk 2 above)
Comment (AbdealiJK) [reply pandas/tests/io/sas/test_sas7bdat.py]: The zero observations file gives a `None` value while reading an empty the csv file gives an empty dataframe. I think this is a bug but have not fixed it right now

**Comment by AbdealiJK** [comment]
@jreback - Thanks for the comments. Updated.

**Comment by AbdealiJK** [comment]
> can you add a whatsnew note for 0.22 (can be in bug fixes). what did this do before with an empty file?

Can you help me with which file I should be updating for this ?

**Comment by jreback** [comment]
doc/source/whatsnew/v0.22.0.txt

**Comment by AbdealiJK** [comment]
@jreback Thanks. its done

File: doc/source/whatsnew/v0.22.0.txt [hunk 4]
```diff
@@ -90,6 +90,7 @@ Bug Fixes
 - Bug in ``pd.read_msgpack()`` with a non existent file is passed in Python 2 (:issue:`15296`)
 - Bug in ``DataFrame.groupby`` where key as tuple in a ``MultiIndex`` were interpreted as a list of keys (:issue:`17979`)
 - Bug in :func:`pd.read_csv` where a ``MultiIndex`` with duplicate columns was not being mangled appropriately (:issue:`18062`)
+- Bug in :func:`pd.read_sas` where a file with 0 variables gives an invalid error. Now it gives and ``EmptyDataError``
```
Comment (jreback) [inline doc/source/whatsnew/v0.22.0.txt]: was the previous error? can you re-word this a bit.

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/whatsnew/v0.22.0.txt (hunk 4 above)
Comment (AbdealiJK) [reply doc/source/whatsnew/v0.22.0.txt]: Done

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @AbdealiJK

**Pull request merged**